compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2515,9 +2515,8 @@ object Parsers {
     def generatorRest(pat: Tree, casePat: Boolean): GenFrom =
       atSpan(startOffset(pat), accept(LARROW)) {
         val checkMode =
-          if (casePat) GenCheckMode.FilterAlways
-          else if sourceVersion.isAtLeast(future) then GenCheckMode.Check
```

Comment:
I'm going to introduce a `3.2-migration` language mode, so should we keep old behaviour under that?

Reply:
even with language modes 3.0 and 3.1 the old behaviour should stand right?

Reply:
yes, I suppose it should. I'll rework it.